### PR TITLE
Server side authentication

### PR DESCRIPTION
**Pull request by aliyorulmazdev**

### CODE REVIEW

File: prExplorer/API/DTOs/RegisterDto.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+namespace API.DTOs
+{
+    public class RegisterDto
+    {
+        [Required]
+        [EmailAddress]
+        public string Email { get; set; }
+        [Required]
+        [RegularExpression("(?=.*\\d)(?=.*[a-z])(?=.*[A-Z]).{4,8}$", ErrorMessage = "Password must be complex")]
```
Comment (dsibinski) [inline prExplorer/API/DTOs/RegisterDto.cs:11]: `Password must be complex`
some more explanation to the user would be useful. Explain to the user what exactly is required for his/her password

File: prExplorer/API/Services/TokenService.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+using System.Text;
+using Domain;
+using Microsoft.IdentityModel.Tokens;
+
+namespace API.Services
+{
+    public class TokenService
+    {
+        // Replace "YourSuperSecretKeyHere" with your actual secret key
+        private const string SecretKey = "0000000000000000000000000000000000000000000000000000000000000000"; // 64 bytes
```
Comment (dsibinski) [inline prExplorer/API/Services/TokenService.cs]: `SecretKey` seems not used, as you fetch it from the config file

File: prExplorer/API/appsettings.Development.json [hunk 3]
```diff
@@ -7,5 +7,6 @@
   },
   "ConnectionStrings": {
     "DefaultConnection": "Data Source=prExplorer.db"
-  }
+  },
+  "TokenKey":"0000000000000000000000000000000000000000000000000000000000000000"
```
Comment (dsibinski) [inline prExplorer/API/appsettings.Development.json:11]: Remember, especially if your code would be open-source, to not expose any keys publicly

File: prExplorer/client-app/src/app/common/form/CustomTextInput.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import React from 'react';
+import { Form, Label } from 'semantic-ui-react';
+
+interface CustomTextInputProps {
+    placeholder: string;
+    name: string;
+    label?: string;
+}
+
+export default function CustomTextInput(customTextInputProps: CustomTextInputProps) {
```
Comment (dsibinski) [inline prExplorer/client-app/src/app/common/form/CustomTextInput.tsx:12]: Isn't this the same code you have already added in another PR? :P

**Review by dsibinski — APPROVED** [review]
@aliyorulmazdev nice, I'm waiting for the login page :)
By the way, did you consider using IdentityServer at all? It's quite standard now in .NET environment and worth knowing. It can be sued for free for up to 1M USD gross revenue :) 

https://duendesoftware.com/products/communityedition

File: prExplorer/client-app/src/app/common/form/CustomTextInput.tsx (hunk 4 above)
Comment (aliyorulmazdev) [reply prExplorer/client-app/src/app/common/form/CustomTextInput.tsx:12]: I changed propsName probally. :)

**Pull request merged**